README.md

```diff
@@ -17,14 +17,36 @@ You can reach the maintainers of this project at:
 
 Participation in the Kubernetes community is governed by the [Kubernetes Code of Conduct](code-of-conduct.md).
 
-## Releases
-
-To reduce version-skew risk, it is recommended to match apiserver-network-proxy server & client
+## Versioning and releases
+
+As of September 2023, the apiserver-network-proxy project is changing its versioning and release
+process. Going forward the project will adhere to these rules:
+
+* The project uses semantic versioning (eg `x.y.z`) for releases and tags.
+* Tags will be created to mark when a release-worthy candidate is ready.
+* Releases will be created by the project maintainers to match tags.
+* Releases and tags follow the Kubernetes minor release version (the `y` in `x.y.z`). For example, when
```

Comment:
* Releases and tags align with Kubernetes minor release versions (y in x.y.z). For instance, if Kubernetes releases version 1.99.0, the corresponding release and tag for apiserver-network-proxy will be 0.99.0